core/src/main/java/com/crazylegend/core/modifiers/SizeTextModifier.kt

```diff
@@ -0,0 +1,133 @@
+import androidx.core.view.setPadding
+import androidx.core.view.updateLayoutParams
+import kotlinx.parcelize.Parcelize
+
+
+/**
+ * Created by crazy on 5/11/20 to long live and prosper !
+ */
+
+
+@Parcelize
+data class SizeTextModifier(
+        var textColor: Int? = null,
+        var textPadding: Int? = null,
+        var textSize: Float? = null,
+        var startMargin: Int? = null,
+        var endMargin: Int? = null,
+        var marginTop: Int? = null,
+        var marginBottom: Int? = null,
+        var margin: Int? = null,
+        var backgroundColor: Int? = null,
+        var backgroundDrawable: Int? = null,
+        var textStyle: TextStyle = TextStyle.NORMAL,
+        var textAlignment: Int = TextView.TEXT_ALIGNMENT_VIEW_START
+) : Parcelable {
+
+
+    private val allSizeMarginCondition get() = margin != null
+
+    enum class TextStyle {
+        BOLD, UNDERLINED, ITALIC, BOLD_ITALIC, NORMAL
+    }
+
+    private fun updateMargins(textView: AppCompatTextView) {
+        textView.updateLayoutParams<ConstraintLayout.LayoutParams> {
+            startMargin?.let { marginStart = it }
+            endMargin?.let { marginEnd = it }
+            marginTop?.let { topMargin = it }
+            marginBottom?.let { bottomMargin = it }
+        }
+    }
+
+    private fun updateMarginsConstraint(textView: AppCompatTextView) {
+        textView.updateLayoutParams<ConstraintLayout.LayoutParams> {
+            startMargin?.let { marginStart = it }
+            endMargin?.let { marginEnd = it }
+            marginTop?.let { topMargin = it }
+            marginBottom?.let { bottomMargin = it }
+        }
+    }
+
+    private fun updateAllMargins(textView: AppCompatTextView) {
+        textView.updateLayoutParams<ConstraintLayout.LayoutParams> {
+            margin?.let { setMargins(it) }
+        }
+    }
+
+    private fun updateAllMarginsConstraint(textView: AppCompatTextView) {
+        textView.updateLayoutParams<ConstraintLayout.LayoutParams> {
+            margin?.let { setMargins(it) }
+        }
+    }
+
+    fun applyTextParams(text: AppCompatTextView) {
+        textSize?.let { text.textSize = it }
+        textPadding?.let { text.setPadding(it) }
+        textColor?.let { text.setTextColor(it) }
+        text.textAlignment = textAlignment
+        backgroundColor?.let { text.setBackgroundColor(it) }
+        applyTextStyle(text)
+        if (allSizeMarginCondition) {
+            updateAllMargins(text)
+        } else {
+            updateMargins(text)
+        }
+    }
+
+    fun applyTextParamsConstraint(text: AppCompatTextView) {
+        textSize?.let { text.textSize = it }
+        textPadding?.let { text.setPadding(it) }
+        textColor?.let { text.setTextColor(it) }
+        text.textAlignment = textAlignment
+        backgroundDrawable?.let { text.background = ContextCompat.getDrawable(text.context, it) }
+                ?: run {
+                    text.backgroundTintList = ColorStateList.valueOf(Color.BLACK)
```

Comment:
what if text color is black and also background tint list is black?

Reply:
Black is the default value. Text will be not visible in that case.

Reply:
If they're the same colors, we should make the one that has weaker luminance darker.

Reply:
should I place check if text color is black then I change the background color to white.

Reply:
Nah, it's much more complex than that but this is an SDK so users should be careful, not our job. LGTM

Reply:
Just remove the dead code and we're ready to merge.

Reply:
Done